provider/query_parameter_value.go

```diff
@@ -0,0 +1,52 @@
+package provider
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Query parameter holds normalized parameter data ready to be inserted in the
+//  final query
+type QueryParameterValue struct {
+	// Token to replace e.g., !TOKEN!
+	Token string
+	// SQL expression to be inserted. Contains "?" that will be replaced with an
+	//  ordinal argument e.g., "$1"
+	SQL string
+	// Value that will be passed to the final query in arguments list
+	Value interface{}
+	// Raw parameter values for debugging and monitoring
+	RawValues map[string]string
```

Comment:
Hm... why is this a map and not just a string? When does it end up with multiple raw values?

Reply:
Hm, it should be a tuple `(string, string)`, but since Go doesn't have tuples in std, I'll split this field in two: `RawParam` and `RawValue`